CASCEdit/Handlers/RootHandler.cs

```diff
@@ -111,58 +111,39 @@ public void RemoveDeleted()
 
 		public void AddEntry(string path, CASResult file)
 		{
-			ulong hash = new Jenkins96().ComputeHash(path);
-			bool found = false;
-
-			// check to see if we're overwriting an existing entry
-			foreach (var root in Chunks)
-			{
-				if (!root.LocaleFlags.HasFlag(locale) && root != GlobalRoot) // skip incorrect locales and non-global roots
-					continue;
+			var cache = CASContainer.Settings.Cache;
 
-				var index = root.Entries.FindIndex(x => x.Hash == hash);
-				if (index >= 0)
-				{
-					RootEntry entry = new RootEntry()
-					{
-						MD5 = file.DataHash,
-						Hash = hash,
-						Path = path,
-						FileDataId = root.Entries[index].FileDataId,
-						FileDataIdOffset = root.Entries[index].FileDataIdOffset
-					};
+			ulong hash = new Jenkins96().ComputeHash(path);
 
-					root.Entries[index] = entry; // update
-					found = true;
+			var entry = Chunks
+						.FindAll(chunk => chunk != GlobalRoot && chunk.LocaleFlags.HasFlag(locale)) // Skip non-global root and incorrect locales
+						.SelectMany(chunk => chunk.Entries) // Select all entries
+						.FirstOrDefault(e => e.Hash == hash); // Filter down to only matching ones
```

Comment:
Should this be a `Where` + iteration instead? IIRC there is the possibility that more than 'root' contains this entry and locale due to different `LocaleFlags` and `ContentFlags` combinations (I may be wrong).

Reply:
Good point. I tried to replicate however this was implemented earlier (see: https://github.com/WowDevTools/CASCHost/pull/3/files#diff-d8e17b05ca7e724ca93972196a702159L120) but technically since we don't yet support specifying other locales/tags we should override all of them. I will check with @Marlamin and implement this if that is the case.